### DISCUSSION TITLE
How to stop exiting the sketch when ESC is pressed?

### DISCUSSION THREAD

**Discussion by mj-mahzoon** [py5coding/py5generator#575] [Q&A] [answered]
In processing, when hitting ESC the sketch will quit. [Here is the code](https://github.com/processing/processing4/blob/85674f8db8c6f91408ca6cb67bb0e97a2a36985b/core/src/processing/core/PApplet.java#L2654C5-L2660C8). Seems like there is a [solution](https://forum.processing.org/one/topic/ignore-escape-key-do-other-action.html) to override this in processing when setting the key to zero. However, I couldn't get this to work in py5. I appreciate if anyone has any suggestions. Thanks!

**Comment by villares** [comment]
Welcome to py5, @mj-mahzoon !

We have been discussing exactly this very recently (https://github.com/py5coding/py5generator/discussions/548) and Jim found it wasn't so easy to make it work in the old Processing way, but on the next release of py5 we'll be able to use an [`intercept_escape()`](http://dev.py5coding.org/reference/sketch_intercept_escape.html) function to handle this!

**Reply by mj-mahzoon** [reply]
Thank you, @villares! Looking forward to it.

**Comment by hx2A** [comment]
Hello, @mj-mahzoon ! As @villares said, the next release will have an `intercept_escape()` method you can use to address this.

I'm currently working to finish up a macOS bug. Once that's done, there will be a release. Hopefully by the end of the year? That's my goal.

**Reply by mj-mahzoon** [reply]
Thanks! Appreciate it!! Any chance that relates to the crash from giving P2D to the size function or losing window focus? :) Looking forward for a fix for those as well!
For the ESC problem, for now I had to created a custom core.jar by removing those lines of code from PApplet.java and re-building the library.

**Reply by hx2A** [reply]
> Any chance that relates to the crash from giving P2D to the size function or losing window focus? :) Looking forward for a fix for those as well!

@mj-mahzoon yes, that's exactly what I am working on. I have a solution that works great on Apple Silicon but I discovered it doesn't work on the older Intel macs. I have a mediocre solution for that but need to iterate on it some more.  I actually plan to start a discussion thread about this to get feedback on some of the design decisions here.

> For the ESC problem, for now I had to created a custom core.jar by removing those lines of code from PApplet.java and re-building the library.

That is a clever workaround! Good job!

**Comment by villares** [comment, marked as answer]
Hurray, solved! https://py5coding.org/reference/sketch_intercept_escape.html

![image](https://github.com/user-attachments/assets/6ea014f8-f081-45cf-9951-ebe2861c424f)
